<a href="https://colab.research.google.com/github/Jeronimocalles/analiticadenegocios/blob/main/Caso_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Caso de Estudio**
Se busca implementar un modelo de árbol de decisión que permita pronosticar si una persona desarrollará enfermedad cardíaca o posee un alto riesgo cardiovascular, a partir de variables clínicas como:

* Edad (Age): que es son los años de vida de la persona al momento de la evaluación.

* Presión sistólica (Systolic): Es la fuerza generada cada vez que el corazón late y bombea sangre a las arterias, y mide la máxima presión cuando el corazón se contrae.

* Presión diastólica (Diastolic): mide la presión mínima cuando el corazón se relaja entre latidos.

* Índice de masa corporal (BMI): , una herramienta médica estándar que evalúa si una persona tiene un peso saludable en relación con su estatura. Se calcula dividiendo el peso en kilogramos por el cuadrado de la altura en metros, para clasificar el bajo peso, peso normal, sobrepeso u obesidad.

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix
from sklearn.model_selection import train_test_split

1. Se procede con la carga de los datos de trabajo

In [ ]:
nxl = '/content/sample_data/2. HeartDisease.xlsx'
XDB = pd.read_excel(nxl, sheet_name=0)

XD = XDB[['Age','Systolic','Diastolic','BMI']]
yd = XDB['Disease']

2. Se procede con la implementación del modelo de árbol

In [ ]:
mar = DecisionTreeClassifier(criterion='gini', max_depth=4)
mar.fit(XD, yd)

ydp = mar.predict(XD)

cm = confusion_matrix(yd, ydp)

VN = cm[0,0]
FP = cm[0,1]
FN = cm[1,0]
VP = cm[1,1]

Exactitud = (VP + VN) / len(XD)
print("La exactitud es: ", Exactitud)

Sensibilidad = VP / (VP + FN)
print("La sensibilidad es: ", Sensibilidad)

Especificidad = VN / (VN + FP)
print("La especificidad es: ", Especificidad)

Precision = VP / (VP + FP)
print("La precisión es: ", Precision)

PrecisionNegativa = VN / (VN + FN)
print("La precisión negativa es: ", PrecisionNegativa)

La exactitud es:  0.8333333333333334
La sensibilidad es:  0.8130841121495327
La especificidad es:  0.863013698630137
La precisión es:  0.8969072164948454
La precisión negativa es:  0.7590361445783133


3. Despliegue del Árbol de Decisión

In [ ]:
from sklearn.tree import export_graphviz
from pydotplus import graph_from_dot_data

vs = ["Age", "Systolic", "Diastolic", "BMI"]

dot_data=export_graphviz(mar, feature_names=vs)
graph=graph_from_dot_data(dot_data)
graph.write_png('Arbol.png')

True

**Análisis de Resultados**

De la base de datos de 180 pacientes se puede observar un total de 90 pacientes que poseen riesgo de enfermedad, mientras que para la categoría de pacientes que no poseen enfermedad es de 90.

Métricas de desempeño Exactitud: 0.8333 (83.33%) → El modelo clasifica correctamente el 83% de los pacientes en general. Sensibilidad: 0.8131 (81.31%) → Detecta correctamente el 81% de los pacientes que sí tienen enfermedad cardíaca. Esto es clave en un contexto médico, ya que reduce el riesgo de dejar pacientes enfermos sin identificar. Especificidad: 0.8630 (86.30%) → Identifica correctamente el 86% de los pacientes sanos. Precisión: 0.8969 (89.69%) → Cuando el modelo predice que un paciente tiene enfermedad cardíaca, acierta casi el 90% de las veces. Predicción Negativa: 0.7590 (75.90%) → Cuando predice que un paciente está sano, acierta aproximadamente el 76% de las veces.

Con respecto a las métricas podemos observar que el modelo logró una exactitud del 83%, lo que indica el buen comportamiento general del modelo frente a la clasificación de los pacientes en el riesgo de tener una enfermedad. Posee una sensibilidad del 81%, que nos da a entender que detecta este pocentaje de los pacientes que si tienen enfermedad cardíaca. Adicionalmente, la especificidad y la precisión, son de 86% y 89% respectivamente, lo que supera el límite inferior definido para este tipo de modelos de clasificación el cual se ubica en el 75%, y nos da entender que el modelo detecta adecuadamente cuando un paciente tiene o no la enfermedad cardíaca.


De acuerdo con el árbol de decisión se destaca un nodo puro (0/23) el cual posee la siguiente regla de decisión: Si una persona cumple con esta tendrá una probabilidad de aprobación del 100%
SI BMI>28 AND AGE>45 AND SYSTOLIC>125 THEN (0/23). Es importante destacar que los nodos puros poseen un gini=0 (el modelo diferencua muy bien los pacientes enfermos).